feat: solve ERR_HTTP_HEADERS_SENT

windforce852 opened the pull request:
card requirement:
Investigate the cause of "Error: Cannot set headers after they are sent to the client,  code: 'ERR_HTTP_HEADERS_SENT'" without global filter when the "email" value is "" in register mutation. and find if there’s any better solution.

completed:
- cause by response.json in RegisterPipeErrorFilter overwrite the response that managed by graphQL and Apollo
- now throw GraphQLError to add error code to response in a graphQL standard rather than REST API standard
- refer to https://www.apollographql.com/docs/apollo-server/migration/#apolloerror

![Screenshot 2024-07-02 at 7 43 36 PM](https://github.com/BeeQuantAI/platform_api/assets/84086904/01234b10-55b7-41cf-9a0e-d4b443dafd7c)

<img width="1459" alt="Screenshot 2024-07-02 at 7 43 44 PM" src="https://github.com/BeeQuantAI/platform_api/assets/84086904/e45ff97f-368f-427e-8727-5b06ce36acad">

Chris-Wang reviewed (changes requested):
This is a good solution, thanks for finding it!

Now you can remove `global-exception.filter` from our project in this PR

Commit:
feat: solve ERR_HTTP_HEADERS_SENT
- Deleted response.json to avoid overwrite to response that managed by graphQL and Apollo
- throw GraphQLError to add error code to response in a graphQL standard rather than REST API standard
- Deleted global filter in app.module.ts as now the RegisterPipeErrorFilter handled the error and will not report error in log
- Deleted global-exception.filter

windforce852 commented:
'global-exception.filter' has now been deleted

Chris-Wang reviewed (approved):
LGTM
